src/main/java/Duke.java

```diff
@@ -1,10 +1,180 @@
+        System.out.println();
+    }
+
+    private static void echo() {
+        String line;
+        Scanner in = new Scanner(System.in);
+        // get user input
+        line = in.nextLine();
+        while (!line.equals("bye")) {
+            System.out.println("    ____________________________________________________________");
+            System.out.println("     " + line);
+            System.out.println("    ____________________________________________________________");
+            line = in.nextLine();
+        }
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Bye. Hope to see you again soon!");
+        System.out.println("    ____________________________________________________________");
+    }
+
+    private static void showList() {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Here are the tasks in your list:");
+        for (int i = 0; i < list.size(); i++) {
+            String description = list.get(i);
+            String time = date.get(i);
+            String category = type.get(i);
+            System.out.printf("     %d.", i + 1);
+            if (done.get(i) == 1) {
+                switch (category) {
+                case "T":
+                    Todo todo = new Todo(description);
+                    System.out.println(todo.doneToString());
+                    break;
+                case "D":
+                    Deadline deadline = new Deadline(description, time);
+                    System.out.println(deadline.doneToString());
+                    break;
+                case "E":
+                    Event event = new Event(description, time);
+                    System.out.println(event.doneToString());
+                    break;
+                default:
+                    System.out.println("       Unknown Object");
+                }
+            } else {
+                switch (category) {
+                case "T":
+                    Todo todo = new Todo(description);
+                    System.out.println(todo);
+                    break;
+                case "D":
+                    Deadline deadline = new Deadline(description, time);
+                    System.out.println(deadline);
+                    break;
+                case "E":
+                    Event event = new Event(description, time);
+                    System.out.println(event);
+                    break;
+                default:
+                    System.out.println("       Unknown Object");
+                }
+            }
+        }
+        System.out.println("    ____________________________________________________________");
+    }
+
+    private static void addList() {
+        String line;
+        Scanner in = new Scanner(System.in);
+        // get user input
+        line = in.nextLine();
+        while (!line.equals("bye")) {
```

Comment:
I think you can consider using functions for each of the if cases to improve the code readability.